mac/ZRTOPTBL.CPY

```diff
@@ -0,0 +1,316 @@
+         C     R5,=AL4(HLASMTBL_END)   * Table overflow?
+         BNL   ERROR05                 * Oh no - need bigger table!
+         MVC   0(L'MARKER,R5),0(R4)    * Copy the entry
+         MVC   LASTOPCD,0(R4)          * update last mnemonic
+         LA    R5,L'MARKER(,R5)        * Bump buffer pointer
+         LA    R4,L'MARKER(,R4)        * Bump input pointer
+         C     R4,=AL4(HLASMREC+1+3*L'MARKER) * record done?
+         BL    GETHLASM_PROCENT        * no: process next entry
+         B     GETHLASM_READ           * yes: get next record
+
+GETHLASM_END   DS 0H
+         ST    R5,HLASMMPT             * Save ptr to empty entry
+         MVC   0(8,R5),=CL8'99999999'  * terminator mnemonic
+         L     R14,GETHLASM_RET        * Get return address
+         BR    R14                     * Return
+
+***********************************************************************
+*
+* Compare routine for extracted opcode definitions
+*
+***********************************************************************
+COMPARE  DS    0H
+         ST    R14,COMPARE_RET          * Save return address
+
+         L     R4,=AL4(Z390TBL)         * Point entries buffer
+         L     R5,Z390MPT               * end of buffer
+         L     R6,=AL4(HLASMTBL)        * Point entries buffer
+         L     R7,HLASMMPT              * Table overflow?
+
+COMPARE_LOOP   DS 0H
+         CLC   0(8,R4),0(R6)            * Compare mnemonics
+         BL    ERROR06                  * Superfluous z390 opcode
+         BH    ERROR07                  * Missing z390 opcode
+         CLC   0(&ENTRYLEN.,R4),0(R6)   * Compare entire entry
+         BNE   ERROR08                  * Report incorrec definition
+         LA    R4,&ENTRYLEN.(,R4)       * Bump z390 entry pointer
+         LA    R6,&ENTRYLEN.(,R6)       * Bump hlasm entry pointer
+COMPARE_NEXT   DS 0H                    * advance to next entry
+         CLR   R4,R5                    * z390 table exhausted?
+         BL    COMPARE_LOOP             * no: compare next entry
+         CLR   R6,R7                    * hlasm table exhausted?
+         BL    COMPARE_LOOP             * no: compare next entry
+
+*
+* We're done: both tables are exhausted
+         L    R14,COMPARE_RET          * Get return address
+         BR   R14                      * Return
+
+***********************************************************************
+*
+* EOF routines
+*
+***********************************************************************
+Z390END  DS   0H
+         OI   Z390EOF,X'FF'            * switch eof indicator on
+         BR   R14
+
+HLASMEND DS   0H
+         OI   HLASMEOF,X'FF'           * switch eof indicator on
+         BR   R14
+
+***********************************************************************
+*
+* Error routines
+*
+***********************************************************************
+ERROR01  DS    0H
+         WTO   'Error: Open failed',ROUTCDE=11,DESC=(6)
+         MVI   RETCODE,X'08'
```

Comment:
Suggestion only. For consistency with other RETCODE settings, use 8 in place of X'08'.